src/shared/misc.c

```diff
@@ -500,3 +509,29 @@ pid_t get_pid(const char *applet,const char *pidfile)
 
 	return pid;
 }
+
+#ifndef HAVE_CLOSE_RANGE
+static inline int close_range(int first RC_UNUSED,
+			      int last RC_UNUSED,
+			      unsigned int flags RC_UNUSED)
+{
+#ifdef SYS_close_range
+	return syscall(SYS_close_range, first, last, flags);
+#else
+	return -1;
```

Comment:
Generally it would be a good idea to set `errno = ENOSYS` before returning.

However, since we don't actually check/log any failures at the call site, I guess it doesn't matter in this specific instance.

Reply:
I agree that `errno = ENOSYS` is a good idea. Let me fix that.

Reply:
Turns out that [syscall(2)](
https://man7.org/linux/man-pages/man2/syscall.2.html) sets `errno`, so we don't need to do it our selves.

> The return value is defined by the system call being invoked.  In general, a 0 return value indicates success.  A -1 return value indicates an error, and an error number is stored in errno.